algorithms/C/strings/reverse a string with spaces using 2d arrays

```diff
@@ -0,0 +1,45 @@
+#include<stdio.h>
+#include<string.h>
+int main()
+{
+    char s1[100][100],s[100];
+    int k=0,j=0,i,n;
+    printf("Enter the string: ");
+    gets(s);                                   // K DENOTES NUMBER OF WORDS
```

Comment:
using `gets ()` along with a fixed length buffer (`s` in this case ) will make the program susceptible to buffer-overflow ( what will happen when input string is of larger size than 100 ? )

either use alternative i/o mechanism ( like `fgets ()` or `getline ()` ) or implement a manual mechanism to check and handle the cases where the input might be overflowing the buffer size.